packages/alfa-rules/src/sia-r83/rule.ts

```diff
@@ -134,29 +139,55 @@ function isPossiblyClipped(device: Device): Predicate<Element> {
   };
 }
 
-function wrapsText(device: Device): Predicate<Text> {
-  return (text) =>
-    text
-      .ancestors({
-        flattened: true,
-      })
-      .filter(isElement)
-      .find(isPossiblyClipped(device))
-      .some((element) => {
-        const style = Style.from(element, device);
-
-        const { value: whitespace } = style.computed("white-space");
-
-        // If whitespace does not cause wrapping, we need to check if a text
-        // overflow could cause the text to clip.
-        if (whitespace.value === "nowrap") {
-          const { value: overflow } = style.computed("text-overflow");
-
-          // We assume that the text won't clip if the text overflow is handled
-          // any other way than clip.
-          return overflow.value !== "clip";
-        }
+function wrapsText(device: Device): Predicate<Element> {
+  return (element) => {
+    if (isPossiblyClipped(device)(element)) {
+      return !isActuallyClipping(element, device);
+    }
+
+    const relevantParent = isPositioned(device, "static")(element)
+      ? element.parent().filter(isElement)
+      : offsetParent(element, device);
+
+    return relevantParent.every(wrapsText(device));
+  };
+}
+
+function isPositioned(device: Device, position: string): Predicate<Element> {
+  return (element) =>
+    Style.from(element, device).computed("position").value.value === position;
+}
```

Comment:
We just got a brand new `hasComputedStyle` 😀 (https://github.com/Siteimprove/alfa/blob/main/packages/alfa-rules/src/common/predicate/has-computed-style.ts)
```suggestion
function isPositioned(device: Device, position: string): Predicate<Element> {
  return hasComputedStyle("position", value => value.value === position, device);
}
```
should do the trick.